spark2/src/main/java/org/apache/iceberg/spark/source/Reader.java

```diff
@@ -156,10 +157,15 @@
     this.io = io;
     this.encryptionManager = encryptionManager;
     this.caseSensitive = caseSensitive;
-
-    this.batchReadsEnabled = options.get("vectorization-enabled").map(Boolean::parseBoolean).orElseGet(() ->
-        PropertyUtil.propertyAsBoolean(table.properties(),
-            TableProperties.PARQUET_VECTORIZATION_ENABLED, TableProperties.PARQUET_VECTORIZATION_ENABLED_DEFAULT));
+    Option<String> option = SparkSession.active().sparkContext().getConf()
+            .getOption("spark.iceberg.vectorization.enabled");
```

Comment:
You should be able to use `SparkSession.active().conf().get("key", null)` to get the value or null. No need to use the Spark context's conf.